### PR TITLE
[navigation-refactor ] Disable KeyboardAvoidingView for the BaseSidebarScreen and fix KeyboardAvoidingView used with react-freeze

### PR DESCRIPTION
**Pull request by adamgrzybowski**
### Details
Disable the KeyboardAvoidingView for the chat list screen. It should prevent the glitch described in the issue.

I couldn't find any use case for this functionality on this screen so I am assuming it's not necessary.

NOTE: `KeyboardAvoidingView` is used only in a file with ios.js extension so these changes shouldn't affect other platforms. For other platforms, it's simply a `View` component.

NOTE 2: There is a patch included so make sure to install it

### Fixed Issues

$ https://github.com/Expensify/App/issues/20514
PROPOSAL: GH_LINK_ISSUE(COMMENT)

### Tests

1. Open iOS app
2. Open a chat
2. Long press a message in chat that doesn't have a thread
3. Tap "Reply in thread" (make sure that the keyboard is visible if you are using simulator, cmd + K)
4. Swipe thread screen to the right to close it and land in chat
5. Swipe chat screen to the right to close it and land in LHN
6. Verify that there is no glitch (part of the list hidden under overlay)

It looks like sometimes you may need to repeat these actions a few times to reproduce this bug, so make sure to test it well.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/67908363/e8785bb6-7601-4a65-88cb-5f99dfae8a87

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/67908363/ce25400b-79b6-4ddc-a9c4-acd5e1deee55

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/67908363/fbb354f4-8680-4c99-970f-ee95819f7887

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/67908363/340d7937-abef-425e-b920-877a60fa1844

</details>

### CODE REVIEW

**Review by mountiny — COMMENTED** [review]
Interesting, thanks for the fix, I have started the build so we can test this on device.

@fedirjh Would you be able to test this one, I think you said you were able to repro in simulator too

**Comment by mountiny** [comment]
Tested the builds and seems to be working well for me on iOS native and mWeb 🎉 

https://github.com/Expensify/App/assets/36083550/634a4988-9bf6-4bba-bbb4-e0d5e2694f70


https://github.com/Expensify/App/assets/36083550/2db20d7d-4158-4221-a02c-c9356ce83e20

**Review by mountiny — DISMISSED** [review]

**Comment by adamgrzybowski** [comment]
Let's wait with this one for a moment. It fixes the problem with the sidebar but @WoLewicki discovered that it may also occur for different screens (e.g. report screen)

We are looking for a more complete solution

**Comment by adamgrzybowski** [comment]
Here you can see the problem that we found https://github.com/Expensify/App/issues/20514#issuecomment-1599233846

**Comment by mountiny** [comment]
thanks for an update!

**Review by mountiny — DISMISSED** [review]
Changes look good to me, SWM will handle creating an upstream PR so we can remove this patch. 

I will have limited availability next week, feel free to merge this once its ready @amyevans 🙇 thank you!

@fedirjh are you able to test this pr thoroughly?

**Comment by adamgrzybowski** [comment]
Hey, I updated the patch with the code that I proposed here https://github.com/facebook/react-native/pull/38074 to fix this upstream

**Comment by adamgrzybowski** [comment]
Hey, @fedirjh
 
> @adamgrzybowski there is a weird behaviour on IOS safari when you navigate back from thread to parent report

Good catch but I think this is a candidate for a separate issue.

It's related to the keyboard but not to the KeyboardAvoidingView. This component is used only in the iOS app. So it may be a little out of the scope of this PR. 

This PR still should fix the problem with the glitch on iOS app

**Comment by adamgrzybowski** [comment]
> @adamgrzybowski Aren’t we disabling the KeyboardAvoidingView for all platforms ?

Yeah, this issue evolved a little 😄 I will update the title of this PR. What we are doing currently is:
- Disable the `KeyboardAvoidingView` for the `BaseSidebarScreen`
- Fix the `KeyboardAvoidingView` so it won't break after freeze (e.g report screen is using freeze)

But the `KeyboardAvoidingView` is used only in the iOS native app.  For other platforms, it's just [`<View>`](https://github.com/Expensify/App/blob/main/src/components/KeyboardAvoidingView/index.js)

What you can observe in the browser is another mechanism. I guess it's just changing the size of the website to make room for the keyboard. But it's not handled by this component.

**Review by fedirjh — APPROVED** [review]
LGTM and tests well.

All yours @amyevans

**Comment by amyevans** [comment]
For the iOS Safari bug identified, do you want me to create a bug issue and assign to you @adamgrzybowski or should we just send it through the regular bug reporting workflow (open to the community for proposals)?

**Review by amyevans — APPROVED** [review]
Code looks good. I tested on both a sim and real device and it tests well.

🚀

**Comment by amyevans** [comment]
> For the iOS Safari bug identified, do you want me to create a bug issue and assign to you @adamgrzybowski or should we just send it through the regular bug reporting workflow (open to the community for proposals)?

@fedirjh want to report it in #expensify-bugs and collect the bounty? We can then assign @adamgrzybowski, or not, based on his answer

File: patches/react-native+0.71.2-alpha.3.patch [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ 
+-    if (this.state.bottom === height) {
++    if (this._bottom === height) {
+       return;
+     }
+ 
+-    if (duration && easing) {
++    this._setBottom(height);
++
++    if (enabled && duration && easing) {
```
[the remark this answers is not in the document]
Comment (adamgrzybowski) [reply patches/react-native+0.71.2-alpha.3.patch:67]: Should I create a new PR for that? @0xmiroslav

**Pull request merged**